src/HOCs/imgixProvider.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useContext, createContext } from 'react'
+
+const ImgixContext = createContext()
+
+/**
+ * `useImgixContext()` tries to invoke `React.useContext()`. If context is 
+ * `undefined`, we know that the context is being accessed outside of a 
+ * provider, and therefore throw an `Error`.
+ * @returns Hook that gets context value from the nearest`ImgixProvider`.
+ */
+function useImgixContext() {
+  const context = useContext(ImgixContext)
+  if (context === undefined) {
+    console.error("useImgixContext must be used within a ImgixProvider")
```

Comment:
This will always log an error when there is no context, right? This probably does more harm than good